Modal de registro para Mailchimp

miglconts opened the pull request:
Hice el modal para el registro pero no estoy entendiendo bien como se monta porque aparece los 2( donar y registro)
ando bien confundido por la gripe aun así, me puedes apoyar explicándome que propiedades tengo que pasarle al modal? lo intente como el de IPFonts

miglconts committed:
Modal de registro para Mailchimp

eduairet commented on frontend/components/Overlay/MailChimp.js:
```diff
@@ -0,0 +1,99 @@
+
+export default function MailChimp() {
+  const [radioValue, setRadioValue] = useState();
+  const emailRef = useRef(null);
```
Estos useRef(null) para qué lo estas usando en este formulario Migue?

miglconts replied:
Para almacenar los valores de los inputs

eduairet replied:
Si, pero el `null` en específico

miglconts replied:
https://es.reactjs.org/docs/hooks-reference.html#useref
Es que así venía en la documentación, se puede usar entonces sin `null`?

eduairet replied:
Si se puede usar sin valor dentro de los paréntesis o puedes pasar un argumento con el valor con el que quieres que se inicie la referencia, en este caso la estás iniciando como null, entonces me dio curiosidad saber el por qué

miglconts replied:
Si, es que es la primera vez que ocupo ese Hook

eduairet replied:
Va :)

eduairet commented on frontend/components/UI/Navbar.js:
```diff
@@ -22,6 +22,7 @@ export default function Navbar() {
                 </Link>
             ))}
             <Button onClick={modalctx.mountHandler}>Donar</Button>
+            <Button onClick={modalctx.mountHandler}>Registro</Button>
```
Aquí puede que te sirva usar el e.target.textContent como el valor que decida qué modal usar
```JSX
mountHandler = e => {
    if (e.target.textContent === 'Donar') { ... }
}
```

eduairet commented on frontend/components/Overlay/MailChimp.js:
```diff
@@ -0,0 +1,99 @@
+                ref={emailRef}
+                required
+              />
+              <label for="firstName-input">Pon tu Nombre</label>
```
Estos inputs podrían ser componentes para limpiar un poco el código, ¿qué opinas?

```JSX
import classes from '../../styles/Form.module.css';

export default function Input(props) {
    return (
        <div className={classes.field}>
            <label className={classes.label} htmlFor={props.id}>
                {props.label}
            </label>
            {props.type !== 'text-area' ? (
                <input
                    className={classes.input}
                    id={props.id}
                    name={props.name}
                    type={props.type}
                    onChange={props.onChange}
                    onBlur={props.onBlur}
                    value={props.value}
                />
            ) : (
                <textarea
                    className={classes.input}
                    id={props.id}
                    name={props.name}
                    onChange={props.onChange}
                    onBlur={props.onBlur}
                    value={props.value}
                ></textarea>
            )}
        </div>
    );
}
```

Así se podrían meter con un map a un arreglo

```JavaScript
const createEditUserInputs = [
    { id: 'name', type: 'text', label: 'Name' },
    { id: 'email', type: 'email', label: 'E-Mail' },
]
```

miglconts replied:
Me parece bien

eduairet replied:
Va Migue :)

miglconts replied:
Intente hacerlo de esta manera pero no logro regresar el `value` desde el componente `Input` al componente `MailChimp`, hice una rama nueva por si quieres checar mi propuesta, quise pasar funciones a traves del `map`, pero no la lee como función

eduairet replied:
Ya te puse unos comentarios en el código del commit de esa rama Migue, a ver si te sirven

miglconts replied:
Voy a estudiarlos pero creo que por el momento los dejare así con el input todo feo

eduairet replied:
Va Migue, está bien, lo importante es que funcione, si ya estás cargando los datos al mailchimp está chido

eduairet reviewed (commented):
- Yo creo que para que logres lo del modal solo necesitas integrar el factor que indique qué modal montar, puedes tener todos los modals en el contexto de modal y solo condicionar cual se monta, tener un prop.currentModal que tenga el text content del modal que se montó y que se limpie cuando se presione el backdrop o cancel.
- ~~Olvidé meter el `package-lock.json` al .gitignore, lo meto para que no nos cause problemas.~~

eduairet commented:
Ya investigué y si es bueno que el package-json-lock se suba! Entonces lo taché en el comentario que había subido

miglconts committed:
Modal Logrado

Use useContext(), un componente que lea una propiedad desde el button del NavBar y renderice el modal adecuado

miglconts commented:
@eduairet ya pude renderizar el modal adecuado, con `useContext()` que lee el valor de `currentModal` que se modifica desde el `button` del `NavBar` con otra función
**Falta darle estilos**

miglconts committed:
try catch

miglconts committed:
Estilos creados || Falta feedback del UI

Solo hace falta agregar una retroalimentación en el modal de registro exitoso

eduairet commented on frontend/components/Overlay/ModalConstructor.js:
```diff
@@ -0,0 +1,18 @@
+      return <MailChimp />;
+
+    default:
+      return null;
```
Quedó chida la lógica montada así

eduairet replied:
Por lectura el ModalConstructor.js podría corresponder al nombre del componente

miglconts replied:
que wey jajaja, bien disléxico mi error, ni me di cuenta

eduairet commented on frontend/components/UI/Navbar.js:
```diff
@@ -1,28 +1,33 @@
+          {navEl.text}
+        </Link>
+      ))}
+      <Button onClick={() => handleClick("Donar")}>Donar</Button>
```
aquí yo creo que podrías pasar la función solo como callback
```JSX
const handleClick = e => {
     modalctx.mountHandler();
     modalctx.setModal(e.target.textContent);
     console.log(modalctx.currentModal);
   };
   return (
     <nav className={classes.nav}>
       {menu.map((navEl, i) => (
         <Link key={`nav-link-${i}`} href={navEl.url}>
           {navEl.text}
         </Link>
       ))}
       <Button onClick={handleClick}>Donar</Button>
```

miglconts replied:
Si lo intente así pero no me jalaba, no se si lo escribia mal o se me iba pasar alguna `prop` pero me pasaba lo mismo que con el conponente `Input` que me recomendaste` y con `() => handleClick("Donar")` me funcionó

eduairet commented on frontend/components/UI/Navbar.js:
```diff
@@ -1,28 +1,33 @@
+        </Link>
+      ))}
+      <Button onClick={() => handleClick("Donar")}>Donar</Button>
+      <Button onClick={() => handleClick("Registro")}>Registro</Button>
```
Lo mismo que arriba, para que mejore la lectura y economizar código

eduairet commented:
Intenté mandar un registro de prueba Migue y hay una falla con un par de módulos

<img width="1019" alt="Screenshot 2022-12-21 at 13 13 16" src="https://user-images.githubusercontent.com/53553463/208985409-b612b20d-2245-4232-8a63-9985c3613e57.png">

eduairet commented:
Creo que los estilos van bien, como este modal que hiciste está dando pauta para los demás yo diría que fuera el estilo que usemos para todos los modals ¿cómo ves?

miglconts commented:
@eduairet instalaste de nuevo las dependencias? 
también tienes que configurar un `.env.local`
De los estilos yo creo que si esta bueno para usarlo de punto de partida

miglconts committed:
Boton arreglado, Modalconstructor arreglado

eduairet commented:
No instalé dependencias 🥸 ya las instalo y configuro el .env.local gracias 🤓

miglconts commented:
Ya arregle el `ModalConstructor` y un error que tenía en botón de registro

miglconts committed:
for cambiados por htmlfor

miglconts committed:
ghtmlFor

miglconts committed:
Pre Registro listo

miglconts committed:
Respuesta en el mismo formulario
